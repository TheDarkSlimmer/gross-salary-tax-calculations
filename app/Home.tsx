import { globalStyles, homeStyles, mainStyles } from "@/styles/styles";
import React, { useState } from "react";
import {
  Image,
  SafeAreaView,
  ScrollView,
  Text,
  TextInput,
  View,
  StatusBar,
  Pressable,
} from "react-native";
import { Link, useLocalSearchParams } from "expo-router";

function Home() {
  const [desiredNetSalary, setDesiredNetSalary] = useState<string>("");
  const [desiredAllowance, setDesiredAllowance] = useState<string>("");
  const { name } = useLocalSearchParams<{ name?: string }>();
  
  return (
    <SafeAreaView>
      <ScrollView>
        <StatusBar backgroundColor={globalStyles.statusColor.backgroundColor} />
        <Text style={homeStyles.helloText}>{`Hello ${name},`}</Text>
        <View style={homeStyles.homeImageContainer}>
          <Image
            source={require("../assets/images/money-income-amico.png")}
            style={{ width: 250, height: 250 }}
            resizeMode="cover"
            alt="Welcome-image"
          />
        </View>
        <View style={homeStyles.homeInputsContainer}>
          <Text style={mainStyles.formDescription}>
            Please enter the details below to get your salary summary
          </Text>
          <TextInput
            keyboardType="numeric"
            placeholder="Enter your desired net salary..."
            style={homeStyles.homeInput}
            onChangeText={(text) => setDesiredNetSalary(text)}
          />
          <TextInput
            keyboardType="numeric"
            placeholder="Enter your total allowances..."
            style={homeStyles.homeInput}
            onChangeText={(text) => setDesiredAllowance(text)}
          />
          <Link
            href={{
              pathname: "/Summary",
              params: {
                salary: parseFloat(desiredNetSalary),
                allowance: parseFloat(desiredAllowance),
              },
            }}
            asChild
          >
            <Pressable
              style={{
                ...globalStyles.buttonStyles,
                backgroundColor:
                  !desiredAllowance || !desiredNetSalary ? "gray" : "#0077B6",
              }}
            >
              <Text style={globalStyles.buttonText}>Submit</Text>
            </Pressable>
          </Link>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
}

export default Home;
