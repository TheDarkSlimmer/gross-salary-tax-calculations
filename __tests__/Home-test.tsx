import React from "react";
import { render } from "@testing-library/react-native";
import Home from "../app/Home";

test("renders correctly", () => {
  const tree = render(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});
