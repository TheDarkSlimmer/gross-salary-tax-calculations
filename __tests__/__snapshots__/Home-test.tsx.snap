// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView>
  <RCTScrollView>
    <View>
      <Text
        style={
          {
            "fontFamily": "Inter-Medium",
            "fontSize": 35,
            "fontWeight": "600",
            "marginBottom": 10,
            "marginLeft": 30,
            "marginTop": "5%",
          }
        }
      >
        Hello undefined,
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
            "marginTop": 25,
            "width": "100%",
          }
        }
      >
        <Image
          alt="Welcome-image"
          resizeMode="cover"
          source={1}
          style={
            {
              "height": 250,
              "width": 250,
            }
          }
        />
      </View>
      <View
        style={
          {
            "display": "flex",
            "gap": 13,
            "justifyContent": "center",
            "marginTop": 30,
            "padding": 30,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "Inter-Light",
              "fontSize": 18,
              "fontWeight": "400",
              "marginTop": 5,
            }
          }
        >
          Please enter the details below to get your salary summary
        </Text>
        <TextInput
          keyboardType="numeric"
          onChangeText={[Function]}
          placeholder="Enter your desired net salary..."
          style={
            {
              "backgroundColor": "#F1F1F1",
              "borderColor": "#787878",
              "borderRadius": 5,
              "borderWidth": 1,
              "fontFamily": "Inter-Light",
              "paddingBottom": 10,
              "paddingLeft": 20,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
        />
        <TextInput
          keyboardType="numeric"
          onChangeText={[Function]}
          placeholder="Enter your total allowances..."
          style={
            {
              "backgroundColor": "#F1F1F1",
              "borderColor": "#787878",
              "borderRadius": 5,
              "borderWidth": 1,
              "fontFamily": "Inter-Light",
              "paddingBottom": 10,
              "paddingLeft": 20,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          href="/Summary?salary=NaN&allowance=NaN"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          role="link"
          style={
            {
              "alignItems": "center",
              "backgroundColor": "gray",
              "borderRadius": 5,
              "borderWidth": 0,
              "display": "flex",
              "height": 38,
              "justifyContent": "center",
              "marginLeft": "auto",
              "marginRight": "auto",
              "width": 140,
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontFamily": "Inter",
                "fontWeight": "600",
              }
            }
          >
            Submit
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
